[SecurityBundle] Add missing `fixXmlConfig()` calls

OskarStark opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 7.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Follows #60929
| License       | MIT

Add `fixXmlConfig()` calls for `'issuers'` and `'algorithms'` arrays to allow using singular XML tags that get automatically converted to plural form. This improves XML configuration ergonomics.

OskarStark committed:
[SecurityBundle] Add missing fixXmlConfig() calls for OIDC token handler arrays

Add fixXmlConfig() calls for 'issuers' and 'algorithms' arrays to allow
using singular XML tags that get automatically converted to plural form.
This improves XML configuration ergonomics.

As requested in PR #60929

xabbuh commented:
Isn't this relevant for older Symfony releases as well?

OskarStark commented:
I would say yes for the issuer and one of the algorithms

stof commented on src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcTokenHandlerFactory.php:
```diff
@@ -126,11 +126,13 @@ public function addConfiguration(NodeBuilder $node): void
                     ->arrayNode('algorithms')
                         ->info('Algorithms used to sign the token.')
                         ->isRequired()
+                        ->fixXmlConfig('algorithm')
```
this is wrong. It needs to be set on the parent

OskarStark replied:
Same for issuer then, right?

stof replied:
yes, all of them

fabpot replied:
@OskarStark Can you make the changes? Having it in 7.4 works for me.

OskarStark commented on src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcTokenHandlerFactory.php:
```diff
@@ -90,7 +90,6 @@ public function addConfiguration(NodeBuilder $node): void
     {
         $node
             ->arrayNode($this->getKey())
```
```suggestion
            ->arrayNode($this->getKey())
                ->fixXmlConfig($this->getKey())
```

OskarStark committed:
Update src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcTokenHandlerFactory.php

OskarStark committed:
Move fixXmlConfig() calls to parent node level

The fixXmlConfig() calls should be placed at the parent node level (oidc)
rather than on individual child array nodes (issuers, algorithms) to properly
handle XML configuration transformations.

fabpot commented on src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcTokenHandlerFactory.php:
```diff
@@ -90,6 +90,8 @@ public function addConfiguration(NodeBuilder $node): void
             ->arrayNode($this->getKey())
                 ->fixXmlConfig($this->getKey())
+                ->fixXmlConfig('issuer')
+                ->fixXmlConfig('algorithm')
```
This one can only work in 8.0 as it already exists in 7.4 (not even sure that this is a good idea as the semantic is not going to be the same between the 2 versions).

OskarStark replied:
See
* https://github.com/symfony/symfony/pull/61702

fabpot commented on src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/AccessToken/OidcTokenHandlerFactory.php:
```diff
@@ -90,6 +90,8 @@ public function addConfiguration(NodeBuilder $node): void
         $node
             ->arrayNode($this->getKey())
                 ->fixXmlConfig($this->getKey())
+                ->fixXmlConfig('issuer')
```
This one is for 7.3.

OskarStark replied:
See 
* https://github.com/symfony/symfony/pull/61701

fabpot reviewed (changes requested):
@OskarStark Can you have a look at my comments and maybe split this PR depending on which version we target?
